src/compas_fab/robots/robot.py

```diff
@@ -1156,6 +1168,15 @@ def inverse_kinematics(self, frame_WCF, start_configuration=None, group=None, re
         >>> robot.inverse_kinematics(frame_WCF, start_configuration, group)                 # doctest: +SKIP
         Configuration((4.045, 5.130, -2.174, -6.098, -5.616, 6.283), (0, 0, 0, 0, 0, 0))    # doctest: +SKIP
         """
+        # Pseudo-memoized sequential calls will re-use iterator if not exhaused
+        request_id = '{}-{}-{}-{}-{}'.format(id(frame_WCF), id(start_configuration), id(group), id(return_full_configuration), id(options))
+
+        if self._current_ik['request_id'] == request_id and self._current_ik['solutions'] is not None:
+            ik = next(self._current_ik['solutions'], None)
+            if ik is not None:
+                joint_positions, joint_names = ik
+                return self._build_configuration(joint_positions, joint_names, group, return_full_configuration)
```

Comment:
Am I reading this correctly?  If the generator is exhausted, then `ik` is `None` and we jump over this `if` clause to then call again `inverse_kinematics`.  Is that really the intuitive behavior?  Especially in the case of analytic kinematics, I would want to know if there were only, say, 4 solutions, and not have to check if the solutions were being repeated.

Reply:
(This situation might come up if we implement the Descartes path planner.)

Reply:
It's a good point. Indeed that is what happens (also added a unit test to verify it). From a user pov, I actually think it is good that IK keeps returning solutions and if using an analytic solver, it cycles back to start. Especially when we're talking about the default use case.  Now, for the cases in which the user actually cares about all possible solutions, this obviously won't cut it. One option would be that we add a new method to the interface, `iter_inverse_kinematics` -which btw, we had a draft PR for ;) - but this time, instead of being a new backend feature, `iter_inverse_kinematics` would simply be the same as `inverse_kinematics` but exposing the generator, so that, if the user needs to, they can get all possible solutions.

How does that sound? Should I add that?

Reply:
mmm... i dunno.  i guess my issue is the different behaviors of the various iks.  `list(robot.inverse_kinematics(...))` causes an infinite loop, but (with your suggestion) `list(client.inverse_kinematics(...))` and `list(robot.iter_inverse_kinematics(...))` wouldn't.  not the worst thing, but also not intuitive/consistent.

Reply:
oh, i retract the previous comment.  `robot.inverse_kinematics` returns things, so wouldn't create an infinite loop, but i guess it's still a similar sort of issue.  the robot's ik has a different return type than the client's...

Reply:
> the robot's ik has a different return type than the client's

yes, that is a bit odd at first, but I guess we both considered that the right thing to do is to keep `Robot.inverse_kinematics`' returning values instead of being a generator. Changing it to a generator would impact a lot of code, and it would make even the simplest use cases more complex than needed. 

So, the main issue that remains is the fact that a same-named method returns different things, right? However, these two methods are already quite different from each other (besides the yield vs return): one outputs configurations, the other joint positions+joint names tuples.

If we change the backend features to have one for `iter_inverse_kinematics` and another for `inverse_kinematics` we would be replicating the situation we had in the draft PR, which was not very convincing, and it adds the need to have extra code in all implementations to handle both. With the proposal of adding `Robot.iter_inverse_kinematics` only, and keeping the rest as is, we would automatically get both behaviors (iterable and single-shot) for all implementations of IK.

Reply:
yeah, `iter_inverse_kinematics` it is then.